src/state/ducks/student/reducer.ts

```diff
@@ -2,43 +2,39 @@ import { createSlice } from "@reduxjs/toolkit";
 import { IStudentRaw } from "./types";
 export const userSlice = createSlice({
   name: "student",
+
   initialState: {
-    data: [
-      {
-        _id: 1,
-        name: "Javed",
-        marks: 100,
-        subject: "Math",
-        grade: "A+",
-        time: "2022-07-18T08:07:28.3Z",
-      },
-      {
-        _id: 2,
-        name: "Ahmad",
-        marks: 80,
-        subject: "English",
-        grade: "B+",
-        time: "2022-07-18T08:07:28.3Z",
-      },
-      {
-        _id: 3,
-        name: "zakria",
-        marks: 40,
-        subject: "English",
-        grade: "F",
-        time: "2022-07-18T08:07:28.3Z",
-      },
-    ],
+    data: [] as IStudentRaw[],
+    specificStudent: {},
   },
 
   reducers: {
     fetchStudentsSuccess: (
       state,
       { payload }: { payload: IStudentRaw[] | any }
     ) => {
-      state.data = [...state.data, ...payload];
+      state.data = [...payload];
+    },
+    deleteStudentSuccess: (
+      state,
+      { payload }: { payload: IStudentRaw[] | any }
+    ) => {
+      const index = state.data.findIndex((obj) => {
+        return obj._id === payload._id;
+      });
+      state.data.splice(index, 1);
```

Comment:
you could do this with filter

state.data = state.data.filter((student)=> student._id !== payload._id)

Also don't use obj as variable, choose a meaninful word.